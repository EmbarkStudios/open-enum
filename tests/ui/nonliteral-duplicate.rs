// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

extern crate open_enum;
use open_enum::open_enum;

// This is a separate test case from duplicate.rs since the error messages
// occur at different points of compilation.

const ONE: isize = 1;

#[open_enum]
enum NonLiteralDuplicateVariant {
    A = 1,
    B = 2,
    C = ONE,
}

#[open_enum]
enum NonLiteralImplicitDuplicateVariant {
    A = ONE,
    B = 0,
    C,
}

fn main() {}
