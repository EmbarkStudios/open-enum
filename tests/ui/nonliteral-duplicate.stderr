error[E0081]: discriminant value `1` assigned more than once
  --> tests/ui/nonliteral-duplicate.rs:24:1
   |
24 | / enum NonLiteralDuplicateVariant {
25 | |     A = 1,
   | |         - first assignment of `1`
26 | |     B = 2,
27 | |     C = ONE,
   | |         --- second assignment of `1`
28 | | }
   | |_^

error[E0081]: discriminant value `1` assigned more than once
  --> tests/ui/nonliteral-duplicate.rs:31:1
   |
31 | / enum NonLiteralImplicitDuplicateVariant {
32 | |     A = ONE,
   | |         --- first assignment of `1`
33 | |     B = 0,
   | |     ----- assigned discriminant for `C` was incremented from this discriminant
34 | |     C,
   | |     - second assignment of `1`
35 | | }
   | |_^
